code/server/src/Server.cpp

```diff
@@ -0,0 +1,104 @@
+#include <Server.h>
+
+bool Server::updateModule() {
+
+    yarp::os::Bottle inputBottle;
+    serverPort.read(inputBottle, true);
+
+    if (isRunning) {
+        if (inputBottle.get(0).asVocab() != COLLATZ_VOCAB_REQ_ITEM) {
+            std::cerr << "Server side: Incorrect Identifier received. Ignoring message," << std::endl;
```

Comment:
in this error condition you somehow break the communication with the client, because you're not sending back any reply